controlplane/kubeadm/internal/cluster.go

```diff
@@ -214,23 +214,67 @@ func (c *cluster) generateEtcdTLSClientBundle() (*tls.Config, error) {
 	}, nil
 }
 
-// etcdIsHealthyResponse is a map of node provider IDs to the etcd health check that failed or nil if no checks failed.
-type etcdIsHealthyResponse map[string]error
+func (c *cluster) getControlPlaneNodes(ctx context.Context) (*corev1.NodeList, error) {
+	controlPlaneNodes := &corev1.NodeList{}
+	controlPlaneNodeLabels := map[string]string{
+		"node-role.kubernetes.io/master": "",
+	}
+
+	if err := c.client.List(ctx, controlPlaneNodes, client.MatchingLabels(controlPlaneNodeLabels)); err != nil {
+		return nil, err
+	}
+	return controlPlaneNodes, nil
```

Comment:
Very minor comment: tall these variables can be simplified, e.g. `controlPlaneNodes -> list` `controlPlaneNodeLabels` -> labels